src/Centreon/Infrastructure/Monitoring/TimelineRepositoryRDB.php

```diff
@@ -0,0 +1,430 @@
+            . $this->generateLogsQuery($collector, $hostId, $serviceId)
+            . ') UNION ALL ('
+            . $this->generateCommentsQuery($collector, $hostId, $serviceId)
+            . ') UNION ALL ('
+            . $this->generateAcknowledgementsQuery($collector, $hostId, $serviceId)
+            . ') UNION ALL ('
+            . $this->generateDowntimesQuery($collector, $hostId, $serviceId)
+            . ')) AS  `event`');
+
+        // Search
+        $searchRequest = $this->sqlRequestTranslator->translateSearchParameterToSql();
+
+        foreach ($this->sqlRequestTranslator->getSearchValues() as $key => $data) {
+            // @todo update when date parameter converter is available
+            $collector->addValue($key, strtotime(current($data)), key($data));
+        }
+        $request .= $searchRequest ? $searchRequest : '';
+
+        // Group
+        $request .= ' GROUP BY event.eventId';
+
+        // Sort
+        $request .= $this->sqlRequestTranslator->translateSortParameterToSql()
+            ?: ' ORDER BY event.timestamp DESC';
+
+        // Pagination
+        $request .= $this->sqlRequestTranslator->translatePaginationToSql();
+
+        $statement = $this->db->prepare($request);
+        $collector->bind($statement);
+
+        $statement->execute();
+
+        $result = $this->db->query('SELECT FOUND_ROWS()');
+        $this->sqlRequestTranslator->getRequestParameters()->setTotal(
+            (int)$result->fetchColumn()
+        );
+
+        while (false !== ($result = $statement->fetch(\PDO::FETCH_ASSOC))) {
+            $newEvent = EntityCreator::createEntityByArray(
+                $this->detectClass($result['eventType']),
+                $result
+            );
+            $timelineEvents[] = new TimelineEvent($newEvent);
+        }
+
+        return $timelineEvents;
+    }
+
+    private function isAdmin(): bool
+    {
+        return ($this->contact !== null)
+            ? $this->contact->isAdmin()
+            : false;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function setContact(ContactInterface $contact): TimelineRepositoryInterface
+    {
+        $this->contact = $contact;
+
+        return $this;
+    }
+
+    /**
+     * @return bool Return FALSE if the contact is an admin or has at least one access group.
+     */
+    private function hasNotEnoughRightsToContinue(): bool
+    {
+        return ($this->contact !== null)
+            ? !($this->contact->isAdmin() || count($this->accessGroups) > 0)
+            : count($this->accessGroups) == 0;
+    }
+
+    /**
+     * Generate SQL query for logs
+     *
+     * @param \Centreon\Infrastructure\CentreonLegacyDB\StatementCollector $collector
+     * @param int $hostId
+     * @param int $serviceId
+     * @return string
+     */
+    protected function generateLogsQuery(StatementCollector $collector, int $hostId, int $serviceId = null): string
+    {
+        $sql = "SELECT
+		CONCAT('L', l.log_id) AS `eventId`,
```

Comment:
IMHO, eventId should be an integer from only eventId
The unique id will be the association between eventType and eventId